src/Angor/Shared/WalletOperations.cs

```diff
@@ -35,7 +36,84 @@ public string GenerateWalletWords()
         string walletWords = mnemonic.ToString();
         return walletWords;
     }
-    
+
+
+    public PsbtWrapper CreatePsbtForTransaction(Transaction transaction, AccountInfo accountInfo, long feeRate, string? changeAddress = null)
+    {
+        Network network = _networkConfiguration.GetNetwork();
+        var nbitcoinNetwork = NetworkMapper.Map(network);
+
+        changeAddress = changeAddress ?? accountInfo.GetNextChangeReceiveAddress();
+
+        var utxoDataWithPaths = FindOutputsForTransaction((long)transaction.Outputs.Sum(_ => _.Value), accountInfo);
+        var coins = utxoDataWithPaths.Select(u => new Coin(uint256.Parse(u.UtxoData.outpoint.transactionId), (uint)u.UtxoData.outpoint.outputIndex, Money.Satoshis(u.UtxoData.value), Script.FromHex(u.UtxoData.scriptHex))).ToList();
+
+        if (!coins.Any())
+            throw new ApplicationException("No coins found to fund the transaction.");
+
+        var builder = new TransactionBuilder(network)
```

Comment:
Are we going to use the transaction builder after all?

Reply:
hmm this was the easy way, I wonder if it is possible to use psbt only, I am going to try.
If we can bypass the builder it will be my preferred option.

Reply:
No easy way, either we use it than have a wrapper for it or we don't use it, otherwise we create more code to fix later on

Reply:
I will wait with this decision, as it is separated to the psbt part.
I rather make sure psbt with taproot works before we start to optimize how we build trx